src/platform/forms-system/src/js/review/ArrayField.jsx

```diff
@@ -226,6 +225,13 @@ class ArrayField extends React.Component {
             const itemSchema = this.getItemSchema(index);
             const itemTitle = itemSchema ? itemSchema.title : '';
 
+            const idSchema = toIdSchema(
+              itemSchema,
+              itemSchema?.$id,
+              this.props.schema.definitions,
+              index,
+            );
```

Comment:
This creates an `idSchema`, and the the `index` arg is used as a suffix that gets appended to each generated id.

Next, we pass this as a prop to `<SchemaForm>`, which passes it as a prop to `<Form>` in the `react-jsonschema-form` library.